cudax/include/cuda/experimental/__data_manipulation/copy.cuh

```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __CUDAX_DATA_MANIPULATION_COPY
+#define __CUDAX_DATA_MANIPULATION_COPY
+
+#include <cuda/__cccl_config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__concepts/__concept_macros.h>
+#include <cuda/std/__ranges/concepts.h>
+
+#include <cuda/experimental/__data_manipulation/common.cuh>
+#include <cuda/experimental/__stream/stream_ref.cuh>
+
+namespace cuda::experimental
+{
+
+template <typename _SrcTy, ::std::size_t _SrcSize, typename _DstTy, ::std::size_t _DstSize>
+void __copy_bytes_impl(
+  stream_ref __stream, _CUDA_VSTD::span<_SrcTy, _SrcSize> __src, _CUDA_VSTD::span<_DstTy, _DstSize> __dst)
```

Comment:
We should not need to pass the size of source and dest because those are dynamically sized